csrc/executor.cpp

```diff
@@ -599,14 +599,14 @@ std::pair<std::vector<int64_t>, std::vector<int64_t>> inferShape(
 // Infer the shape of an intemediate tensor using kir::Allocate
 std::pair<std::vector<int64_t>, std::vector<int64_t>> inferShapeOfIntermediate(
     const TensorView* tv,
-    const kir::Allocate* alloc,
     ExpressionEvaluator& expr_eval) {
-  // Allocate should be provided for intermediates. We just need to
-  // grab a chunk of memory of the size dicatated by
-  // Allocate::shape().
-  TORCH_INTERNAL_ASSERT(alloc != nullptr);
+  auto alloc_dom = TensorDomain::noReductions(tv->getMaybeAllocationDomain());
```

Comment:
Shouldn't this also ignore broadcast domains? See `getGlobalAllocationSizes` in lower_allocation.cpp

Reply:
Removed broadcasts.